### ISSUE TITLE
Tutorial 4: Why not set the padding_idx of nn.Embedding's to src_pad_token_idx or tgt_pad_token_idx respectively?

### ISSUE THREAD

**Issue by yipliu** [bentrevett/pytorch-seq2seq#167]
Hi Ben,

I notice the Parameters **padding_idx** in **nn.Embedding**. After I read the description, I think it will work for Tutorial 4.

**_padding_idx (int, optional)_** – If specified, the entries at padding_idx do not contribute to the gradient; therefore, the embedding vector at padding_idx is not updated during training, i.e. it remains as a fixed “pad”. For a newly constructed Embedding, the embedding vector at padding_idx will default to all zeros, but can be updated to another value to be used as the padding vector.

**Comment by bentrevett** [comment]
They can be set, yes, and I have done this for some other NLP projects like text classification - however, I have found that it never makes any noticable difference so introduces additional complexity at no benefit, so I've decided to ignore it here.

Feel free to try it for the tutorials and letting me know if it makes a difference or not in your projects. If it does I will add it to the tutorials.
